calico_containers/adapter/datastore.py

```diff
@@ -590,6 +596,87 @@ def remove_ip_pool(self, version, pool):
             raise KeyError("%s is not a configured IP pool." % pool)
 
     @handle_errors
+    def get_hostnames(self):
+        """
+        Return a list of hostnames
+        :return:  A list of hostnames configured in etcd.
+        """
+        # Get a list of the hosts.
+        hosts = []
+        results = self.etcd_client.read(HOSTS_PATH)
+        for result in results.children:
+            packed = result.key.split("/")
+            # Handle no hosts being configured (etcd returns the parent).
+            if len(packed) != 5:
+                continue
+            hosts.append(packed[-1])
+        return hosts
+
+    @handle_errors
+    def get_node_as_peer(self, version, hostname):
```

Comment:
I found this name very confusing, since "as" = Autonomous System.  Can we rename this?

Also, I don't think it's a very appropriate method to become part of the public interface to datastore.py.  You only really want it as a sub-procedure of generating the "mesh" peers.  So we should give it a leading `_`.

I know you use it in calicoctl.py, but only as a test of whether or not the node's own BGP config exists.  However, that's actually an error condition an we should just throw an exception in get_node_to_node_mesh_peers.